src/include/tpp.h

```diff
@@ -94,6 +92,115 @@ extern "C" {
 
 #endif
 
+#define	RPP_RETRY	10
+/*
+ **	Default allowed number of outstanding pkts.
+ */
+#define	RPP_HIGHWATER	1024
+
+/*
+ * Default number of RPP packets to check every server iteration
+ */
+#define RPP_MAX_PKT_CHECK_DEFAULT	64
```

Comment:
so we have two server attrs (aka `rpp_highwater` and `rpp_retry`) and these macros are the helper of those attrs.
Since there is no change for those attrs, so I have kept this code unmodified...
We can do two things here:
1. Rename those attrs to more generic name (or like tpp_highwater and tpp_retry (I personally don't like these name, but couldn't think better name now), But that will be external change and may affect user who is using these attrs
2. Keep rpp_highwater and rpp_retry external name as it but in code change everything to tpp_ along with all helper macros and funcs, and eventually rename external name also

But I am not sure about any of the above so, I kept those unchanged.

Reply:
That is a good question.  what do these two attributes do?  Maybe we can just get rid of them if they were only rpp related.  RPP was all about us retrying sending packets.  Doesn't rpp_retry have to do with that?  Now that we're not having to resend any packets, it seems redundant.  I'm not sure I ever quite understood what the highwater attribute did.

Reply:
Even I don't understand them fully, maybe @subhasisb can help here... But currently few places those attrs are still used, even in TPP also.

Reply:
We still do retries and resends in TPP as well, though rarely used. SInce the connections go via a router daemon, these are virtual connections - e.g. the connection between say server and one mom is not a direct TCP connection. There is a TCP connection between server and comm and then another connection from comm to mom. If two comms are configured (for failover), the sending side keeps the packet till it receives an ack from the receiver. If that does not arrive in time,  then the sending party will send the message over the secondary comm. While doing this logic i reused the rpp_retry and rpp_highwater variables